tests/test_plugin.py

```diff
@@ -932,3 +932,84 @@ def test_hook_chaining(node_factory):
     assert(l2.daemon.is_in_log(
         r'plugin-hook-chain-even.py: htlc_accepted called for payment_hash {}'.format(hash3)
     ))
+
+
+def test_bitcoin_backend(node_factory, bitcoind):
+    """
+    This tests interaction with the Bitcoin backend, but not specifically bcli
+    """
+    l1 = node_factory.get_node(start=False, options={"disable-plugin": "bcli"},
+                               may_fail=True, allow_broken_log=True)
+
+    # We don't start if we haven't all the required methods registered.
+    plugin = os.path.join(os.getcwd(), "tests/plugins/bitcoin/part1.py")
+    l1.daemon.opts["plugin"] = plugin
+    try:
+        l1.daemon.start()
+    except ValueError:
+        assert l1.daemon.is_in_log("Missing a Bitcoin plugin command")
+        # Now we should start if all the commands are registered, even if they
+        # are registered by two distincts plugins.
+        del l1.daemon.opts["plugin"]
+        l1.daemon.opts["plugin-dir"] = os.path.join(os.getcwd(),
+                                                    "tests/plugins/bitcoin/")
+        try:
+            l1.daemon.start()
+        except ValueError:
+            msg = "All Bitcoin plugin commands registered"
+            assert l1.daemon.is_in_log(msg)
+        else:
+            raise Exception("We registered all commands but couldn't start!")
+    else:
+        raise Exception("We could start without all commands registered !!")
+
+    # But restarting with just bcli is ok
+    del l1.daemon.opts["plugin-dir"]
+    del l1.daemon.opts["disable-plugin"]
+    l1.start()
+    assert l1.daemon.is_in_log("bitcoin-cli initialized and connected to"
+                               " bitcoind")
+
+
+def test_bcli(node_factory, bitcoind):
+    """
+    This tests the bcli plugin, used to gather Bitcoin data from a local
+    bitcoind.
+    Mostly sanity checks of the interface..
+    """
+    l1, l2 = node_factory.get_nodes(2)
+
+    # We cant stop it dynamically
+    with pytest.raises(RpcError):
+        l1.rpc.plugin_stop("bcli")
+
+    # Failure case of feerate is tested in test_misc.py
+    assert "feerate" in l1.rpc.call("getfeerate", {"blocks": 3,
+                                                   "mode": "CONSERVATIVE"})
+
+    resp = l1.rpc.call("getchaininfo")
+    assert resp["chain"] == "regtest"
```

Comment:
This needs mocking as well since we may be running on `liquid-regtest`.

Reply:
Ah forgot this one thanks!